### PR TITLE
[react-native] Add missing style props to View

### PR DESCRIPTION
**Pull request by brunolemos**
Continuation of #25083 

React Native actually accepts the whole `ViewStyle` as props, not only the `FlexStyle`.

Before this PR, it wouldn't allow me to do `<View backgroundColor="#00FF00" />`, after it will.

> ~PS: I noticed an existing type error but I don't know how to fix it: `overflow` can have the `scroll` value, but it is currently not autocompleting that. If we add this the `ImageStyle` gives an error when trying to override it.~ (fixed)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/master/Libraries/Components/View/ViewStylePropTypes.js
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by iRoachie — APPROVED** [review]

File: types/react-native-snap-carousel/index.d.ts [hunk 1]
```diff
@@ -180,11 +179,11 @@ export interface CarouselProps<T> extends React.Props<ScrollViewProps> {
     /**
      * Optional styles for Scrollview's global wrapper
      */
-    containerCustomStyle?: StyleProp<ScrollViewStyle>;
+    containerCustomStyle?: StyleProp<ViewStyle>;
```
Comment (alloy) [inline types/react-native-snap-carousel/index.d.ts]: I’m a little confused, isn’t this limiting the available styles to a superset of the available scrollview styles?

**Comment by alloy** [comment]
I’d really appreciate some changes to the tests that reflect the important changes in here to ensure we can fully understand them now and in the future.

**Review by alloy — CHANGES_REQUESTED** [review]
I’d really appreciate some changes to the tests that reflect the important changes in here to ensure we can fully understand them now and in the future.

File: types/react-native-snap-carousel/index.d.ts (hunk 1 above)
Comment (iRoachie) [reply types/react-native-snap-carousel/index.d.ts]: It's supposed to be a ViewStyle according to the source https://github.com/archriss/react-native-snap-carousel/blob/59b1d90f7e5ad23d4c4ced241ec15379e5d2fb6d/src/carousel/Carousel.js#L46

Comment (alloy) [reply types/react-native-snap-carousel/index.d.ts]: Ah gotcha 👍

**Comment by brunolemos** [comment]
I can add a couple tests later today

**Comment by alloy** [comment]
Thanks! 👍

**Review by alloy — APPROVED** [review]

**Comment by hpmax00** [comment]
#27595

**Pull request merged**